test/extended/two_node/tnf_kubelet_disruption.go

```diff
@@ -0,0 +1,284 @@
+
+		nodes = nodeList.Items
+		framework.Logf("Found nodes: %s and %s for kubelet disruption test", nodes[0].Name, nodes[1].Name)
+
+		kubeClient := oc.KubeClient()
+		etcdClientFactory = helpers.NewEtcdClientFactory(kubeClient)
+
+		g.By("Ensuring both nodes are healthy before starting kubelet disruption test")
+		for _, node := range nodes {
+			o.Eventually(func() bool {
+				return isNodeReady(oc, node.Name)
+			}, nodeIsHealthyTimeout, pollInterval).Should(o.BeTrue(), fmt.Sprintf("Node %s should be ready before kubelet disruption", node.Name))
+		}
+
+		g.By("Validating cluster is in good state before kubelet disruption")
+		o.Eventually(func() error {
+			return validateClusterOperatorsAvailable(oc)
+		}, etcdOperatorIsHealthyTimeout, pollInterval).ShouldNot(o.HaveOccurred(), "All cluster operators should be available before kubelet disruption")
+	})
+
+	g.It("Should recover from single node kubelet service disruption", func() {
+		targetNode := nodes[0]
+		survivingNode := nodes[1]
+		var constraintId string
+
+		g.By(fmt.Sprintf("Adding constraint to avoid kubelet resource on node: %s", targetNode.Name))
+		err := addConstraint(oc, survivingNode.Name, "kubelet-clone", targetNode.Name)
+		o.Expect(err).To(o.BeNil(), fmt.Sprintf("Expected to add constraint for kubelet resource on node %s without errors", targetNode.Name))
+
+		g.By("Discovering the constraint ID for later removal")
+		constraintId, err = discoverConstraintId(oc, survivingNode.Name, "kubelet-clone", targetNode.Name)
+		o.Expect(err).To(o.BeNil(), "Expected to discover constraint ID without errors")
+		framework.Logf("Discovered constraint ID: %s", constraintId)
+
+		g.By("Waiting for target node to become NotReady due to kubelet constraint")
+		o.Eventually(func() bool {
+			return !isNodeReady(oc, targetNode.Name)
+		}, kubeletDisruptionTimeout, kubeletPollInterval).Should(o.BeTrue(), fmt.Sprintf("Node %s should become NotReady after kubelet constraint is applied", targetNode.Name))
+
+		g.By(fmt.Sprintf("Ensuring surviving node %s remains Ready during kubelet disruption", survivingNode.Name))
+		o.Consistently(func() bool {
+			return isNodeReady(oc, survivingNode.Name)
+		}, 2*time.Minute, kubeletPollInterval).Should(o.BeTrue(), fmt.Sprintf("Surviving node %s should remain Ready during kubelet disruption", survivingNode.Name))
+
+		g.By("Validating etcd cluster remains healthy with surviving node")
+		o.Eventually(func() error {
+			return helpers.EnsureHealthyMember(g.GinkgoT(), etcdClientFactory, survivingNode.Name)
+		}, kubeletDisruptionTimeout, pollInterval).ShouldNot(o.HaveOccurred(), fmt.Sprintf("etcd member %s should remain healthy during kubelet disruption", survivingNode.Name))
+
+		g.By(fmt.Sprintf("Removing constraint (ID: %s) to allow kubelet resource on node: %s", constraintId, targetNode.Name))
+		err = removeConstraint(oc, survivingNode.Name, constraintId)
```

Comment:
If the test fails before this line it never removes the constraint and the next tests will fail as well. Consider add AfterEach to ensure constraints are always removed.